src/main/java/com/treasuredata/client/TDHttpClient.java

```diff
@@ -450,38 +464,48 @@ private TDClientException handleHttpResponseError(String apiRequestPath, int cod
                     String conflictsWith = errorResponse.isPresent() ? parseConflictsWith(errorResponse.get()) : null;
                     throw new TDClientHttpConflictException(errorMessage, conflictsWith);
                 case HttpStatus.PROXY_AUTHENTICATION_REQUIRED_407:
-                    throw new TDClientHttpException(PROXY_AUTHENTICATION_FAILURE, errorMessage, code);
+                    throw new TDClientHttpException(PROXY_AUTHENTICATION_FAILURE, errorMessage, code, retryAfter);
```

Comment:
I don't get the policy whether to pass `retryAfter` to these exceptions or not since 401, 404 or 409 isn't set `retryAfter` while others are set. Also, maybe we don't need to pass `retryAfter` when the exception isn't 408, 429 nor 5xx.